src/main/java/com/dataloom/search/requests/SearchDetails.java

```diff
@@ -0,0 +1,42 @@
+package com.dataloom.search.requests;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import com.dataloom.client.serialization.SerializationConstants;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class SearchDetails implements Serializable {
+    private static final long serialVersionUID = -6691065251875288358L;
+
+    private final String searchTerm;
+    private final UUID propertyType;
+    private final boolean exactMatch;
+    
+    @JsonCreator
+    public SearchDetails(
+            @JsonProperty( SerializationConstants.SEARCH_TERM ) String searchTerm,
+            @JsonProperty( SerializationConstants.PROPERTY_FIELD ) UUID propertyType,
+            @JsonProperty( SerializationConstants.EXACT ) boolean exactMatch ) {
+        this.searchTerm = searchTerm;
```

Comment:
This as true as long as no server code (or client Java code) ever instantiates this object. The `checkNotNull(...)` is more of a defensive thing since you can't be sure how the object will be used once you make the library.  The rare case for an exception is in an identified performance bottleneck where tracing has identified it as significant bottleneck.